### PR TITLE
[sovereign] Cross chain header storage bootstrap

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Extended header cleanup and apply notarized data from storage on bootstrap
  
## Proposed changes
- Overwrite `applyCrossNotarizedHeaders`, `cleanupNotarizedStorage`, `cleanupNotarizedStorageForHigherNoncesIfExist` for `sovereignChainShardStorageBootstrapper` 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go
Comment (axenteoctavian) [inline process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go:1]: In each `log.Debug` you print hash as `[]byte`, isn't it better to use `hex.EncodeToString`?

Comment (mariusmihaic) [reply process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go:1]: No need, logger takes care of that

File: process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go [hunk 1]
```diff
@@ -19,5 +25,118 @@ func NewSovereignChainShardStorageBootstrapper(shardStorageBootstrapper *shardSt
...
+	if !castOk {
+		log.Warn("sovereignChainShardStorageBootstrapper.cleanupNotarizedStorage",
+			"error", process.ErrWrongTypeAssertion,
+			"expected shard header of type", "SovereignChainHeaderHandler",
+		)
+		return
+	}
+
+	for _, extendedHeaderHash := range sovereignHeader.GetExtendedShardHeaderHashes() {
+		var extendedHeader data.HeaderHandler
```
Comment (sasurobert) [inline process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go:76]: this line can be moved outside of the for.

File: process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go [hunk 2]
```diff
@@ -19,5 +25,118 @@ func NewSovereignChainShardStorageBootstrapper(shardStorageBootstrapper *shardSt
...
+		nonce++
+
+		extendedBlock, extendedBlockHash, err := process.GetExtendedHeaderFromStorageWithNonce(
+			nonce,
+			ssb.store,
+			ssb.uint64Converter,
+			ssb.marshalizer,
+		)
+		if err != nil {
+			log.Debug("extended block is not found in ExtendedShardHeadersNonceHashDataUnit storage",
```
Comment (sasurobert) [inline process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go:118]: can you rewrite this ? like only in case of error do we have an exit from the for. This does not look good. Can you rephrase ?

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go (hunk 1 above)
Comment (mariusmihaic) [reply process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go:76]: Right, moved it outside

File: process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go (hunk 2 above)
Comment (mariusmihaic) [reply process/sync/storageBootstrap/sovereignChainShardStorageBootstrapper.go:118]: Yes, it is the same way as in `shardStorageBootstrapper` -> basically you cleanUp anything until there is nothing more to cleanUp.

I will rephrase the log, though

**Review by sasurobert — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]
